config/serialization.go

```diff
@@ -169,7 +176,7 @@ func Save(config map[string]map[string]interface{}) error {
 	}
 
 	// Write config file
-	configFile := configFile()
+	_ ,configFile := configFile()
```

Comment:
If this is the spot you were talking about requiring filename, you can make the config write function receive an `io.Writer` and make `configFile` return `io.ReadWriterCloser`